ts/features/idpay/initiative/details/components/BeneficiaryDetailsContent.tsx

```diff
@@ -82,13 +83,30 @@ const BeneficiaryDetailsContent = (props: BeneficiaryDetailsProps) => {
     O.getOrElse(() => "-")
   );
 
-  const rewardPercentageString = pipe(
-    beneficiaryDetails.refundRule?.accumulatedAmount?.refundThreshold,
-    O.fromNullable,
-    O.map(percentage => `${percentage}%`),
-    O.getOrElse(() => "-")
-  );
-
+  const getRewardTypeDependantString = () => {
+    if (
+      initiativeDetails.initiativeRewardType === InitiativeRewardTypeEnum.REFUND
+    ) {
+      return pipe(
+        beneficiaryDetails.refundRule?.accumulatedAmount?.refundThreshold,
+        O.fromNullable,
+        O.map(percentage => `${percentage}%`),
+        O.getOrElse(() => "-")
+      );
+    }
+    return pipe(
+      beneficiaryDetails.rewardRule?.rewardValue,
+      O.fromNullable,
+      O.fold(
+        () => "-",
+        value =>
+          beneficiaryDetails.rewardRule?.rewardValueType ===
+          RewardValueTypeEnum.PERCENTAGE
+            ? `${value}%`
+            : "-" // will be added once the ABSOLUTE reward type is supported
+      )
+    );
+  };
```

Comment:
The problem here is that we used the wrong value. So it should be enough to do this:
```suggestion
  const rewardPercentageString = pipe(
    beneficiaryDetails.rewardRule?.rewardValue,
    O.fromNullable,
    O.map(percentage => `${percentage}%`),
    O.getOrElse(() => "-")
  );
```
Both initiative types use this value currently. We will add the absolute value when there will the need (if ever).

Reply:
in case of a `REFUND` initiative, we go checking the re**FUND**Rule field, while in case of a `REWARD`(aka discount) type initiative, we go checking the re**WARD**rule one, else we'd have the same problem that this PR is trying to fix (if I got it right).
are you sure about your correction? from reading the DTOs, this looked like the correct solution to me; remember this screen is used for both discount and refund initiatives

Reply:
Both DTOs use the `rewardRule` object in this section. Refund is never expressed in percentage and refers to **when** the refund will be executed, and not to **how much** it will be

Reply:
adressed in [6d8707f](https://github.com/pagopa/io-app/pull/4719/commits/6d8707fe9e87e2b03d7f05bdd342d5bcd2583739)